src/Audio_details.md

```diff
@@ -0,0 +1,192 @@
+
+[^digital_analog]:
+To be clear: digital values are discrete and clear-cut; conversely, the analog domain is continuous.
+The former is what computers use, the latter is what the real world is made of.
+
+[^vin]:
+Actually, VIN acts as a 5<sup>th</sup> channel fed into the mixer, whose control bits are in NR50 instead of NR51.
+This was omitted from the diagram for simplicity.
+
+## PCM registers
+
+These two registers, only present in the Game Boy Color and ulterior models, allow reading the output of the generation circuits directly; this is very useful for testing "internal" APU behavior.
+These registers are not documented in any known Nintendo manual.
+
+### FF76 — PCM12 (CGB Mode only): Digital outputs 1 & 2 \[read-only\]
+
+The low nibble is a copy of sound channel 1's digital output, the high nibble a copy of sound channel 2's.
+
+### FF77 — PCM34 (CGB Mode only): Digital outputs 3 & 4 \[read-only\]
+
+Same, but with channels 3 and 4.
+
+## Finer technical explanation
+
+### DIV-APU
+
+A "DIV-APU" counter is increased every time `DIV`'s bit 4 (5 in [double-speed mode](<#FF4D — KEY1 (CGB Mode only): Prepare speed switch>)) goes from 1 to 0, therefore at a frequency of 512 Hz (regardless of whether double-speed is active).
+Thus, the counter can be made to increase faster by writing to `DIV` while its relevant bit is set (which clears `DIV`, and triggers the falling edge).
+
+Some events are tied to the DIV-APU counter:
+
+Event          | Every N ticks | Frequency[^div_apu_freq]
+---------------|---------------|-------------------------
+Envelope sweep | 8             | 64 Hz
+Sound length   | 2             | 256 Hz
+CH1 freq sweep | 4             | 128 Hz
+
+[^div_apu_freq]:
+Indicated values are under normal operation; the frequencies will obviously differ if writing to `DIV` to increase the counter faster.
+
+### Mixer
+
+A high-pass filter (HPF) removes constant biases over time.
+The HPFs therefore remove the DC offset created by inactive channels with an enabled DAC, and off-center waveforms.
+
+Enabling or disabling a DAC ([see below](#DACs)), adding or removing it using NR51, or changing the volume in NR50, will cause an audio pop.
+(All of these actions cause a change in DC offset, which is smoothed out by the HPFs over time, but still creates a pop.)
+
+The HPF is more aggressive on GBA than on GBC, which itself is more aggressive than on DMG.
+(The more "aggressive" a HPF, the faster it pulls the signal towards "analog 0"; this reduces the pops, but tends to also distort waveforms.)
+
+### DACs
+
+Channel <var>x</var>'s DAC is enabled if and only if `[NRx2] & $F8 != 0`; the exception is CH3, whose DAC is directly controlled by bit 7 of NR30 instead.
+Note that the envelope functionality changes the volume, but not the value stored in NRx2, and thus doesn't disable the DACs.
+
+If a DAC is enabled, the digital range $0 to $F is lineraly translated to the analog range -1 to 1, in arbitrary units.
```

Comment:
What does "in arbitrary units" wants to communicate here? Aren't we in arbitrary units already in the digital range?

Reply:
Yes, but those are different unary units, and the usual unit for analog signals is volts, and we don't want to imply that sound ranges from -1 V to 1 V. I'm not sure the exact values are known, and they would muddy the actual explanations throughout the page.

Reply:
I don't see V specified anywhere else here when speaking about analog ranges and/or signals. I think it's clear enough to refer to them without specifying units.

Reply:
When reading documents just stating values without units, I default to assuming that it's a typo/oversight, and try to figure out what the unit is. Saying it explicitly clears that up.